### PR TITLE
[Chat] Updated Teams Identity Logic

### PR DESCRIPTION
**Pull request by jpeng-ms**
# What

updated how teams identity is determined for file cards

# Why

improvements

# How Tested

CI snapshots

# Process & policy checklist

- [x] I have updated the project documentation to reflect my changes if necessary.
- [x] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

**Review by jimchou-dev — APPROVED** [review]
Assume identifying teams bot user will be in separate PR, LGTM

File: packages/acs-ui-common/src/identifier.ts [hunk 1]
```diff
@@ -60,3 +60,10 @@ export const _isValidIdentifier = (identifier: CommunicationIdentifier): boolean
     isUnknownIdentifier(identifier)
   );
 };
+
+/**
+ * Check if given identifier is a Microsoft Teams user.
+ *
+ * @internal
+ */
+export const isIdentityMicrosoftTeamsUser = isMicrosoftTeamsUserIdentifier;
```
Comment (emlynmac) [inline packages/acs-ui-common/src/identifier.ts:69]: Why are we wrapping this rather than just using the library directly?

Comment (emlynmac) [reply packages/acs-ui-common/src/identifier.ts:69]: There are some wrapping functions above showing how underlying functionality can be combined to provide more useful functions. They also have the preceding underscore in the name to mark them as internal.

File: packages/react-components/src/components/Attachment/AttachmentDownloadCards.tsx [hunk 2]
```diff
@@ -152,7 +152,8 @@ const getDefaultMenuActions = (
 ): AttachmentMenuAction[] => {
   let actionName = locale.openAttachment;
   // if message is sent by a Teams user, we need to use a different icon ("open")
-  if (chatMessage?.senderId?.includes('8:orgid:')) {
+  const isTeamsUser = isIdentityMicrosoftTeamsUser(_toCommunicationIdentifier(chatMessage?.senderId ?? ''));
```
Comment (emlynmac) [inline packages/react-components/src/components/Attachment/AttachmentDownloadCards.tsx:155]: This implicitly brings in `CommunicationIdentifier` as a return type and we should probably have the underlying wrapper function in the common area handle this.
`_isIdentityMicrosoftTeamsUser(value: string?) -> bool`

**Review by emlynmac — APPROVED** [review]

**Pull request merged**